accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -291,6 +291,17 @@ The following claims are required to fully support generating build provenance f
 | Build reference | Ref to an external workflow if one is being used, e.g. a trusted builder. | job_workflow_ref | N/A                                     | N/A                                                            |
 
 ## Implementation details
+
+There will be two different ways to include provenance information during publish: running the CLI directly on a supported CI/CD system or using the optional but recommended "trusted builder to publish.
+
+- "Sigstore integration in the npm CLI"
+  - The npm CLI will include the capability to extract provenance information out-of-the-box on supported CI/CD systems.
+  - The provenance information will be extracted from the ENV and ID token provided by the CI/CD system when running `npm publish`.
+- "Non-falsifiable provenance using a trusted builder"
+  - A reusable workflow or "trusted builder" can be used to invoke the npm CLI, extracting provenance, running install, build and publish in isolated steps, preventing malicious dependencies or build script from tampering with the provenance information.
+  - The trusted builder will for example checkout the repo that's in the ID token as well as extract ENV information before any other user supplied commands are executed, passing this information to `npm publish` for signing.
+  - Using the trusted builder is recommended but it might not support all possible publish workflows so will be optional.
```

Comment:
Does this mean that i can provide provenance information when building and publishing on my local machine?

Reply:
Not in this current proposal 😞  using the trusted builder will be optional when publishing from CI/CD.

I'm not opposed to supporting "author/maintainer" based identities for signing in future, for example using an email based identity or your npm account to sign the build. This would then allow you to publish from anywhere and bind a particular package to the maintainer.

This would be more like an extension to this proposal. I'd like to focus on build or workflow identities on CI/CD systems as a first step though to keep things manageable.